src/javascript/JContent/ContentRoute/ContentLayout/ContentLayout.gql-queries.js

```diff
@@ -183,6 +183,9 @@ class FilesQueryHandler {
                                 }
                                 children(typesFilter: {types: ["jnt:resource"]}) {
                                     nodes {
+                                        data: property(name: "jcr:data") {
```

Comment:
Looks like you are retrieving the whole binary. That will likely have a huge impact on performance right?
The length should likely be computed on the server side instead (if not stored anywhere).

Reply:
ok.i'll take a look